frontend/components/puzzle2.jsx

```diff
@@ -1,20 +1,46 @@
 import React from 'react';
+import { Route, Redirect } from 'react-router-dom';
 
 // Answer: See o double you
-
 class Puzzle2 extends React.Component {
-  render() {
 
-    return (
-      <div>
+ constructor(props) {
+    super(props);
+    
+    this.state = {
+      puzzlesolved: 'no',
+      blankanswer: 'no',
+      wronganswer: 'no'
+    }
+ };
+   submitanswer() {
+      var userinput = document.getElementById('puzzleAnswer').value;
+      if (userinput) {
+      this.props.send('submit', {puzzle: '1', answer: userinput})
+      console.log('answer was submitted')
+  } else {
+    this.setState({blankanswer : 'yes'})
+    console.log('there was a blank answer')
+  }
+}
+  
+   
+render() {
+   //if statement for incorrect answers
+   if (this.props.incorrectanswer === true) {
+     document.getElementById("puzzleAnswer").style.borderColor = "red";
```

Comment:
Two options are better than this in my opinion:
One is to put a style prop on the input element `style={{ borderColor: red }}`
Second is to use a styled component and pass in `incorrectanswer` as a prop. This helps separate the JSX from the styles, but still keeps them co-located in the same file.

Reply:
so this style prop would be defined and become a part of this.state? Will need to do more research on this as well xD

Reply:
mm, not part of `this.state`, but using `this.props.incorrectanswer` to determine the value
so for option one: `style={{ borderColor: this.props.incorrectanswer ? 'red' : 'transparent' }}`
or for option two:
```
import styled from 'styled-components';

const AnswerAwareInput = styled.input`
  borderColor: ${props => props.incorrectanswer ? 'red' : 'transparent'};
`;
```
and then using `AnswerAwareInput` in place of `input`. See `frontend/components/main.jsx` for more examples of styled components

Reply:
went with option 2